### PR TITLE
feat(ingest): utilities for query logs

### PR DESCRIPTION
**Pull request by hsheth2**
- Make it configurable via an env variable
- Add a CLI for extracting the query logs from the sqlite db
- Add a test for the stored query logs
- Fix broken data lake profiling test
- Bump acryl-sqlglot

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]

File: metadata-ingestion/src/datahub/cli/check_cli.py [hunk 1]
```diff
@@ -339,3 +343,28 @@ def test_path_spec(config: str, input: str, path_spec_key: str) -> None:
...
+@click.argument("query-log-file", type=click.Path(exists=True, dir_okay=False))
+@click.option("--output", type=click.Path())
+def extract_sql_agg_log(query_log_file: str, output: Optional[str]) -> None:
+    """Convert a sqlite db generated by the SqlParsingAggregator into a JSON."""
+
+    from datahub.sql_parsing.sql_parsing_aggregator import LoggedQuery
+
+    assert dataclasses.is_dataclass(LoggedQuery)
+
+    shared_connection = ConnectionWrapper(pathlib.Path(query_log_file))
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/cli/check_cli.py:358]: Does this need to get closed?

File: metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py [hunk 2]
```diff
@@ -281,6 +313,9 @@ def __init__(
             assert self.usage_config is not None
             self._usage_aggregator = UsageAggregator(config=self.usage_config)
 
+    def close(self) -> None:
+        self._exit_stack.close()
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py:317]: Nice

File: metadata-ingestion/src/datahub/utilities/file_backed_collections.py [hunk 3]
```diff
@@ -126,6 +126,7 @@ def executemany(
     def close(self) -> None:
         for obj in self._dependent_objects:
             obj.close()
+        self._dependent_objects.clear()
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/utilities/file_backed_collections.py:129]: Nice

**Review by asikowitz — APPROVED** [review]
Nice bug fixes on file backed data structures. One question

File: metadata-ingestion/src/datahub/cli/check_cli.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/cli/check_cli.py:358]: probably would be nicer to do it explicitly, but it'll get done in `__del__` so it's fine

**Pull request merged**